.github/workflows/sanity.yaml

```diff
@@ -0,0 +1,22 @@
+name: sanity
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    paths:
+      - '**'
+
+jobs:
+  sanity:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '~1.16'
+      - name: Run formatting checks
+        run: make vendor && make format && git diff --exit-code
+      - name: Ensure codegen is up-to-date
+        run:  make generate && make manifests && git diff --exit-code
```

Comment:
can we use the `verify` make target for this instead?

how about adding specific targets that verify each thing we care about; e.g. `codegen`, `manifests`, `goimports`, etc?